usaspending_api/search/v2/views/search.py

```diff
@@ -493,39 +496,14 @@ def post(self, request):
 
                 return Response(district_response)
 
-    def state_results_matview(self, filter_args, lookup_fields, loc_lookup):
-        # Adding additional state filters if specified
-        if self.geo_layer_filters:
-            self.queryset = self.queryset.filter(**{'{}__{}'.format(loc_lookup, 'in'): self.geo_layer_filters})
-        else:
-            # Adding null filter for state for specific partial index
-            # when not using geocode_filter
-            filter_args['{}_{}'.format(loc_lookup, 'isnull')] = False
-
-        self.geo_queryset = self.queryset.filter(**filter_args) \
-            .values(*lookup_fields) \
-            .annotate(federal_action_obligation=Sum('federal_action_obligation'))
-
-        # State names are inconsistent in database (upper, lower, null)
-        # Used lookup instead to be consistent
-        results = [
-            {
-                'shape_code': x[loc_lookup],
-                'aggregated_amount': x['federal_action_obligation'],
-                'display_name': code_to_state.get(x[loc_lookup], {'name': 'None'}).get('name').title()
-            } for x in self.geo_queryset
-        ]
-
-        return results
-
     def state_results(self, filter_args, lookup_fields, loc_lookup):
         # Adding additional state filters if specified
         if self.geo_layer_filters:
             self.queryset = self.queryset.filter(**{'{}__{}'.format(loc_lookup, 'in'): self.geo_layer_filters})
         else:
             # Adding null filter for state for specific partial index
             # when not using geocode_filter
-            filter_args['{}__{}'.format(loc_lookup, 'isnull')] = False
+            filter_args['{}_{}'.format(loc_lookup, 'isnull')] = False
```

Comment:
should be '__' since isnull is not a location field. throwing error when there are are no geo_layer filters for states